### ISSUE TITLE
git.latest state with remote set fails on first try

### ISSUE THREAD

**Issue by ChristianBeer** [saltstack/salt#42690]
### Description of Issue/Question
When adding a second remote to an existing repository the first try fails with a python error. The remote is added but not set as remote.

The stacktrace points to https://github.com/saltstack/salt/blob/develop/salt/states/git.py#L1145 where it seems that `fetch_url` is `None` when it should not be.

### Setup
Simple config to illustrate issue:
```
repository-create:
  git.latest:
    - name: "https://github.com/octocat/Spoon-Knife.git"
    - target: "/srv/test/repository/."
    - rev: master
    - branch: master

repository-add-remote:
  git.latest:
    - name: "https://github.com/ChristianBeer/Spoon-Knife.git"
    - target: "/srv/test/repository/."
    - remote: downstream
    - rev: master
    - branch: master
    - require:
      - git: repository-create
```

### Steps to Reproduce Issue
minion log:
```
2017-08-02 16:01:03,403 [salt.loaded.int.module.cmdmod][ERROR   ][12884] Command '['git', 'rev-parse', '--show-toplevel']' failed with return code: 128
2017-08-02 16:01:03,403 [salt.loaded.int.module.cmdmod][ERROR   ][12884] stderr: fatal: Not a git repository (or any of the parent directories): .git
2017-08-02 16:01:03,403 [salt.loaded.int.module.cmdmod][ERROR   ][12884] retcode: 128
2017-08-02 16:01:05,874 [salt.loaded.int.states.git][ERROR   ][12884] Unexpected exception in git.latest state
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/states/git.py", line 1144, in latest
    salt.utils.url.redact_http_basic_auth(fetch_url),
  File "/usr/lib/python2.7/dist-packages/salt/utils/url.py", line 196, in redact_http_basic_auth
    return re.sub(url_re, redacted, output, flags=re.IGNORECASE)
  File "/usr/lib/python2.7/re.py", line 155, in sub
    return _compile(pattern, flags).sub(repl, string, count)
TypeError: expected string or buffer
2017-08-02 16:01:05,876 [salt.state       ][ERROR   ][12884] expected string or buffer
```

### Versions Report
```
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Jan 19 2017, 14:48:08)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.1
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-3-amd64
         system: Linux
        version: debian 9.1
```

**Comment by gtmanfred** [comment]
I have attempted to fix this in https://github.com/saltstack/salt/pull/42694

Can you check that?

Thanks,
Daniel

**Comment by ChristianBeer** [comment]
I downloaded the changed `git.py` and replaced the one installed on my Debian 9 VM with salt 2017.7 and it fixes the problem. Thank you for this quick response.

My real use case is adding a remote without making it the default remote which currently is only possible by ordering the states so that the default remote is executed last but that always creates two changes with every highstate because every `git.latest` state switches the default remote. I'll open a feature request for that.

**Comment by gtmanfred** [comment]
Sounds good, thanks!

Daniel

**Comment by ChristianBeer** [comment]
Will the fix be added to the 2017 and future releases too? The current pull request just fixes it in 2016.11.

**Comment by gtmanfred** [comment]
It will be forward merged to 2017.7 and then into develop

https://docs.saltstack.com/en/latest/topics/development/contributing.html#the-current-release-branch

**Comment by ChristianBeer** [comment]
Thanks for the link.
